src/main/java/org/veupathdb/service/eda/us/model/UserDataFactory.java

```diff
@@ -376,6 +367,77 @@ public void transferGuestAnalysesOwnership(long fromGuestUserId, long toRegister
     );
   }
 
+  /***************************************************************************************
+   *** Read analysis metrics
+   **************************************************************************************/
+
+  public UserAnalysisMetricsResponse readAnalysisMetrics(Date startDate, Date endDate) {
+    UserAnalysisMetricsResponseImpl response = new UserAnalysisMetricsResponseImpl();
+    response.setStartDate(startDate);
+    response.setStartDate(endDate);
+    response.setCreationCounts(getAnalysisCountsPerDateType(startDate, endDate, "creation_time"));
+    response.setModifiedCounts(getAnalysisCountsPerDateType(startDate, endDate, "modification_time"));
+    return response;
+  }
+
+  UserAnalysisCounts getAnalysisCountsPerDateType(Date startDate, Date endDate, String dateTypeColumn) {
+
+    UserAnalysisCounts counts = new UserAnalysisCountsImpl();
+
+    // All analyses in this time period
+    List<StudyCount> allAnalyses =  readAnalysisCountsByStudy(startDate, endDate, dateTypeColumn, false);
+    counts.setAnalysesCountPerStudy(allAnalyses);
+    int allCount = allAnalyses.stream().map(StudyCount::getCount).reduce(0, Integer::sum);
+    counts.setTotalAnalysesCount(allCount);
+
+    // Imported analyses in this time period
+    List<StudyCount> importedAnalyses =  readAnalysisCountsByStudy(startDate, endDate, dateTypeColumn, true);
+    counts.setImportedAnalysesPerStudy(importedAnalyses);
+    int importedCount = importedAnalyses.stream().map(StudyCount::getCount).reduce(0, Integer::sum);
+    counts.setTotalImportedAnalysesCount(importedCount);
+
+    return counts;
+  }
+
+  public List<StudyCount> readAnalysisCountsByStudy(Date startDate, Date endDate, String dateColumn, boolean limitToImports) {
+    String sqlTemplate = """
+select count(analysis_id) as cnt, study_id
+from %sanalysis
+where %s > ? and %s <= ?
+%s
+group by study_id
+order by cnt desc
+        """;
+    String importClause = limitToImports? "and provenance is not null" + System.lineSeparator(): "";
+    String sql = String.format(sqlTemplate, _userSchema, dateColumn, dateColumn, importClause);
+
+    java.sql.Date sqlStartDate = java.sql.Date.valueOf(LocalDate.of(1990, 1, 1)); // epoch start
```

Comment:
Did you check that you actually need to do this conversion (java.util.Date -> java.sql.Date)?  Not sure you do.  Also, can do a ternary for each of these rather than assign a default before you know you need to.